From ccc188fcf98ad9166ee551fac9d94b2603c3a51b Mon Sep 17 00:00:00 2001
From: Ihor Radchenko <yantar92@posteo.net>
Date: Tue, 20 Feb 2024 12:43:51 +0300
Subject: [PATCH] lisp/files.el (untrusted-content): New variable

The new variable is to be used when buffer contents comes from untrusted
source. This is a prerequisite for CVE-2024-53920 and CVE-2026-96442 fixes
to work correctly - trusted-content-p reads this variable.

Backport of upstream commit ccc188fcf98ad9166ee551fac9d94b2603c3a51b.
---
 lisp/files.el | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lisp/files.el b/lisp/files.el
index aaaaaaa..bbbbbbb 100644
--- a/lisp/files.el
+++ b/lisp/files.el
@@ -577,6 +577,14 @@ Non-nil means enable use of directory-local variables.
 Some modes may wish to set this to nil to prevent directory-local
 settings being applied, but still respect file-local ones.")
 
+(defvar-local untrusted-content nil
+  "Non-nil means that current buffer originated from an untrusted source.
+Email clients and some other modes may set this non-nil to mark the
+buffer contents as untrusted.
+
+This variable might be subject to change without notice.")
+(put 'untrusted-content 'permanent-local t)
+
 ;; This is an odd variable IMO.
 ;; You might wonder why it is needed, when we could just do:
 ;; (set (make-local-variable 'enable-local-variables) nil)
-- 
2.39.0
